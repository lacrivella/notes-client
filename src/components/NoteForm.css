@import url('https://fonts.googleapis.com/css?family=Press+Start+2P&display=swap');

.NoteForm {
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  font-family: 'Press Start 2P', cursive;

  button {
    height: 5rem;
    border-radius: 8px;
    margin: 1rem;
  }

  input {
    text-align: center;
    margin: 1rem;
    height: 2rem;
    width: 15rem;
  }

  textarea {
    height: 5rem;
    width: 15rem;
    align-items: center
    ;
  }
}