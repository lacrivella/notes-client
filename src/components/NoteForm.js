import React from 'react';
import PropTypes from 'prop-types';
import styles from './NoteForm.css';

function NoteForm({ title, body, handleTitleChange, handleBodyChange, handleSubmit }) {
  return (
    <section className={styles.NoteForm}>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={title}
          placeholder="enter title"
          onChange={handleTitleChange} />
        <textarea
          type="text"
          value={body}
          placeholder="enter task"
          onChange={handleBodyChange} />
        <button>Add to Tasks</button>
      </form>
    </section>
  );
}

NoteForm.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleTitleChange: PropTypes.func.isRequired,
  handleBodyChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default NoteForm;
