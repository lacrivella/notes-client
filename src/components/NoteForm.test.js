import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

describe('NoteForm component', () => {
  it('renders NoteForm', () => {
    const wrapper = shallow(<NoteForm 
      title="Title"
      handleTitleChange={() => { }}
      body="body"
      handleBodyChange={() => { }}
      handleSubmit={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
