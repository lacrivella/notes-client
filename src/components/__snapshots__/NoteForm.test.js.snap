// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm component renders NoteForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="enter title"
    type="text"
    value="Title"
  />
  <textarea
    onChange={[Function]}
    placeholder="enter task"
    type="text"
    value="body"
  />
  <button>
    Add to Tasks
  </button>
</form>
`;
